prov/gni/test/api.c

```diff
@@ -247,6 +242,11 @@ static void rdm_api_teardown_common(bool unreg)
 		fi_freeinfo(hints[i]);
 	}
 
+	free(uc_source);
+	free(uc_target);
+	free(target);
+	free(source);
+
```

Comment:
Because we're freeing the memory before closing the MR.  (Commit message includes this).